command/command.ts

```diff
@@ -1750,6 +1761,9 @@ export class Command<
       const args = this.parseArguments(ctx, options);
       this.literalArgs = ctx.literal;
 
+      //
+      // TODO: How can we defer executing this when ctx.execute is false?
+      //
       // Execute option action.
       if (ctx.actions.length) {
```

Comment:
This is the problematic part that is executed only in the parse flow. I haven't given it much thought, but this think probably needs to be deferred somehow and/or unified with the `execute` method so it's consistently handled.